azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineScaleSetApiMockTest.java

```diff
@@ -0,0 +1,116 @@
+   }
+
+   public void testVMSSGet_NetworkProfile() throws InterruptedException {
+      server.enqueue(jsonResponse("/virtualmachinescaleset.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").properties().virtualMachineProfile()
+         .networkProfile()
+         .networkInterfaceConfigurations()
+         .size(), 1);
+   }
+
+   public void testVMSSCreate_BaseProperty() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").location(), "eastus");
+   }
+
+   public void testVMSSCreate_SKUProperty() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").sku().name(), "Standard_A1");
+   }
+
+   public void testVMSSCreate_Property() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").properties().singlePlacementGroup().booleanValue(), true);
+   }
+
+   public void testVMSSCreate_StorageProfile() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").properties().virtualMachineProfile()
+         .storageProfile()
+         .osDisk()
+         .managedDiskParameters()
+         .storageAccountType().toString(), "Standard_LRS");
+   }
+
+   public void testVMSSCreate_PropertyUpgradePolicy() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").properties().upgradePolicy().mode(), "Manual");
+   }
+
+   public void testVMSSCreate_NetworkProfile() throws InterruptedException {
+      server.enqueue(jsonResponse("/createvirtualmachinescalesetresponse.json").setResponseCode(200));
+      final VirtualMachineScaleSetApi vmssAPI = api.getVirtualMachineScaleSetApi(resourcegroup);
+      assertEquals(vmssAPI.get("jclouds-vmssname").properties().virtualMachineProfile()
+         .networkProfile()
+         .networkInterfaceConfigurations()
+         .size(), 1);
+   }
```

Comment:
These mock tests still don't meet their purpose:

* There should be only **one** test for each method in the API to verify that the method generates the expected HTTP request. Verifying that means the test method should:
  - [x] Validate the returned object (this verifies serialization)
  - [ ] Validate the generated requests using the `assertSent` method. This validates that method invocation generated the expected HTTP request (expected URI, headers, and body).
* If the API method defines a `@Fallback`, add another test for that method that exercises it.
  - [ ] For all methods with fallbacks, add a test that returns the response code that triggers the fallback, and verify that the method returns the expected result (usually `null`, empty lists, etc) instead of throwing an exception.

In summary, this test class should just have the following test methods: `testList`, `testListWhen404`, `testGet`, `testGetWhen404`, `testCreateOrUpdate`, `testDelete` and `testDeleteWhen404`.

The [AvailabilitySetApiMockTest](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/AvailabilitySetApiMockTest.java) is a good example to follow.